csharp/source/autowrap/csharp/csharp.d

```diff
@@ -0,0 +1,1061 @@
+    rangeDef.functions ~= "        [DllImport(\"%s\", EntryPoint = \"autowrap_csharp_slice_Bool_Slice\", CallingConvention = CallingConvention.Cdecl)]".format(libraryName) ~ newline;
+    rangeDef.functions ~= "        internal static extern slice dlang_slice_Bool_Slice(slice dslice, IntPtr begin, IntPtr end);" ~ newline;
+    rangeDef.functions ~= "        [DllImport(\"%s\", EntryPoint = \"autowrap_csharp_slice_Bool_AppendValue\", CallingConvention = CallingConvention.Cdecl)]".format(libraryName) ~ newline;
+    rangeDef.functions ~= "        internal static extern slice dlang_slice_Bool_AppendValue(slice dslice, [MarshalAs(UnmanagedType.Bool)] bool value);" ~ newline;
+    rangeDef.functions ~= "        [DllImport(\"%s\", EntryPoint = \"autowrap_csharp_slice_Bool_AppendSlice\", CallingConvention = CallingConvention.Cdecl)]".format(libraryName) ~ newline;
+    rangeDef.functions ~= "        internal static extern slice dlang_slice_Bool_AppendSlice(slice dslice, slice array);" ~ newline;
+    rangeDef.constructors ~= "            if (typeof(T) == typeof(bool)) this._slice = RangeFunctions.dlang_slice_Bool_Create(new IntPtr(capacity));" ~ newline;
+    rangeDef.enumerators ~= "                else if (typeof(T) == typeof(bool)) yield return (T)(object)RangeFunctions.dlang_slice_Bool_Get(_slice, new IntPtr(i));" ~ newline;
+    rangeDef.getters ~= "                else if (typeof(T) == typeof(bool)) return (T)(object)RangeFunctions.dlang_slice_Bool_Get(_slice, new IntPtr(i));" ~ newline;
+    rangeDef.setters ~= "                else if (typeof(T) == typeof(bool)) RangeFunctions.dlang_slice_Bool_Set(_slice, new IntPtr(i), (bool)(object)value);" ~ newline;
+    rangeDef.sliceEnd ~= "            else if (typeof(T) == typeof(bool)) return new Range<T>(RangeFunctions.dlang_slice_Bool_Slice(_slice, new IntPtr(begin), _slice.length));" ~ newline;
+    rangeDef.sliceRange ~= "            else if (typeof(T) == typeof(bool)) return new Range<T>(RangeFunctions.dlang_slice_Bool_Slice(_slice, new IntPtr(begin), new IntPtr(end)));" ~ newline;
+    rangeDef.appendValues ~= "            else if (typeof(T) == typeof(bool)) { range._slice = RangeFunctions.dlang_slice_Bool_AppendValue(range._slice, (bool)(object)value); return range; }" ~ newline;
+    rangeDef.appendArrays ~= "            else if (typeof(T) == typeof(bool)) { range._slice = RangeFunctions.dlang_slice_Bool_AppendSlice(range._slice, source._slice); return range; }" ~ newline;
+
+    generateRangeDef!byte(libraryName);
+    generateRangeDef!ubyte(libraryName);
+    generateRangeDef!short(libraryName);
+    generateRangeDef!ushort(libraryName);
+    generateRangeDef!int(libraryName);
+    generateRangeDef!uint(libraryName);
+    generateRangeDef!long(libraryName);
+    generateRangeDef!ulong(libraryName);
+    generateRangeDef!float(libraryName);
```

Comment:
Then it needs to be enforced programatically if it can.

Reply:
What needs to be enforced here?

Reply:
`//This needs to be written last`

Reply:
It is enforced insofar as the method is private and it is called last in the C# generation process. I can remove the comment if that removes any confusion.

Reply:
Oh. I interpreted it as a comment for someone trying to wrap C# code, not for the implementation. I'd just add a why then.

Reply:
I wrote it as a note to myself back in the early days of the code. Looking at it now, it doesn't even apply any more. Removed.